internal/pkg/composable/providers/host/host.go

```diff
@@ -92,11 +85,16 @@ func (c *contextProvider) Run(comm corecomp.ContextProviderComm) error {
 	}
 }
 
-func onFQDNFeatureFlagChange(fqdnFFChangeCh chan struct{}) features.BoolValueOnChangeCallback {
-	return func(new, old bool) {
-		// FQDN feature flag was toggled, so notify on channel
-		fqdnFFChangeCh <- struct{}{}
-	}
+func (c *contextProvider) onFQDNFeatureFlagChange(new, old bool) {
+	// FQDN feature flag was toggled, so notify on channel
+	c.fqdnFFChangeCh <- struct{}{}
```

Comment:
See comment below about the buffer channel.

You only need to ensure that a struct is on the channel, if one is already there you really don't need to push it. So I would change this line to:

```
select {
case c.fqdnFFChangeCh <- struct{}{}:
default:
}
```

This removes any chance of blocking, but ensures that if Run is running or will be ran it will get the notification.

Reply:
Thanks.  Addressed in 4a5a7c2cebe7bfe77637319ba1e12b2b4ea63cd2.